src/weatherreport/src/weatherreport_check_custodian.erl

```diff
@@ -0,0 +1,81 @@
+%% -------------------------------------------------------------------
+%%
+%% weatherreport - automated diagnostic tools for CouchDB
+%%
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%%
+%% @doc Diagnostic that performs safety and liveness checks on
+%% cluster shards. Shard safety is determined by the availability of
+%% the nodes that contain copies of that shard. A shard is considered
+%% unsafe if one or more nodes containing copies are unavailable.
+%% Shard liveness is similar but also requires nodes containing copies
+%% to be actively participating in the cluster. If one or more nodes
+%% containing copies are in maintenance mode then liveness is impaired.
+%% Messages are also returned for any databases where there are
+%% conflicting shard maps.
+
+-module(weatherreport_check_custodian).
```

Comment:
I thought this was coming out? Unless Cloudant intends to donate custodian, too.

Reply:
Custodian has [already been donated and imported into 3.x](https://github.com/apache/couchdb/commit/15c8243f3ba21bac8c05374293b79813f1461f96).